Gradle build fails

Gradle won't build anything for me, apparently because I don't have an `ANDROID_HOME` env. Output is:

```
FAILURE: Build failed with an exception.

* Where:
Build file '/mnt/data/codes/Java/settlers-remake/go.graphics.android/build.gradle' line: 15

* What went wrong:
A problem occurred evaluating project ':go.graphics.android'.
> SDK location not found. Define location with sdk.dir in the local.properties file or with an ANDROID_HOME environment variable.
```

Also, my eclipse workspace doesn't build anymore. What do I have to do to be able to build the swing target again?

1. I dont have gradle. I use the gradlew.bat in the root folder instead. On the website they told me I should use it if one is shipped with the project.
2. It tells me:
```
λ gradlew :jsettlers.main.swing:run --info
Starting Build
Settings evaluated using settings file 'C:\master\settings.gradle'. << This file does not exist
Projects loaded. Root project using build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle'.
Included projects: [root project 'settlers-remakeH']
Evaluating root project 'settlers-remakeH' using build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle'.
Compiling build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle' using SubsetScriptTransformer.
Compiling build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle' using BuildScriptTransformer.

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle' line: 92

* What went wrong:
A problem occurred evaluating root project 'settlers-remakeH'.
> Task with path ':jsettlers.mapcreator:run' not found in root project 'settlers-remakeH'.

BUILD FAILED

Total time: 7.677 secs
Stopped 0 compiler daemon(s).
```
If I comment line 92 out, the error is in line 93, line 124...

3.I found an android sdk on my drive. Created a local.properties file:
`failed to find target with hash string 'android-15' in: C:\Program Files (x86)\Android\android-sdk`
But I have an platforms/android-19 folder?!

1. I dont have gradle. I use the gradlew.bat in the root folder instead. On the website they told me I should use it if one is shipped with the project.
2. It tells me:
```
λ gradlew :jsettlers.main.swing:run --info
Starting Build
Settings evaluated using settings file 'C:\master\settings.gradle'. << This file does not exist
Projects loaded. Root project using build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle'.
Included projects: [root project 'settlers-remakeH']
Evaluating root project 'settlers-remakeH' using build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle'.
Compiling build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle' using SubsetScriptTransformer.
Compiling build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle' using BuildScriptTransformer.

FAILURE: Build failed with an exception.

* Where:
Build file 'C:\Users\...\Projects\Java\settlers-remakeH\build.gradle' line: 92

* What went wrong:
A problem occurred evaluating root project 'settlers-remakeH'.
> Task with path ':jsettlers.mapcreator:run' not found in root project 'settlers-remakeH'.

BUILD FAILED

Total time: 7.677 secs
Stopped 0 compiler daemon(s).
```
If I comment line 92 out, the error is in line 93, line 124...

3.I found an android sdk on my drive. Created a local.properties file:
`failed to find target with hash string 'android-15' in: C:\Program Files (x86)\Android\android-sdk`
But I have an platforms/android-19 folder?!

--- a/build.gradle
+++ b/build.gradle
@@ -91,23 +91,25 @@ tasks.getByPath(":jsettlers.network:run").onlyIf { !gradle.taskGraph.hasTask(":r
 
 task runMapCreator(dependsOn: ':jsettlers.mapcreator:run') {}
 
-task runAndroid(dependsOn: ':jsettlers.main.android:assembleDebug') << {
-    Properties local = new Properties()
-    local.load(new FileInputStream("local.properties"))
-    def adb = local.getProperty('sdk.dir') + '/platform-tools/adb'
-    println 'Copy files to your device...'
-    exec {
-        commandLine adb, 'uninstall', 'jsettlers.main.android'
-    }
-    exec {
-        commandLine adb, 'install', '-r', './jsettlers.main.android/build/outputs/apk/jsettlers.main.android-debug.apk'
-    }
-    exec {
-        commandLine adb, 'shell', 'am', 'start', '-n', 'jsettlers.main.android/jsettlers.main.android.JsettlersActivity'
-    }
-    println 'The application should have started now.'
+if (findProject(':jsettlers.main.android')) {
+	// Android is present.
+	task runAndroid(dependsOn: ':jsettlers.main.android:assembleDebug') << {
+	    Properties local = new Properties()
+	    local.load(new FileInputStream("local.properties"))
+	    def adb = local.getProperty('sdk.dir') + '/platform-tools/adb'
+	    println 'Copy files to your device...'
+	    exec {
+	        commandLine adb, 'uninstall', 'jsettlers.main.android'
+	    }
+	    exec {
+	        commandLine adb, 'install', '-r', './jsettlers.main.android/build/outputs/apk/jsettlers.main.android-debug.apk'
+	    }
+	    exec {
+	        commandLine adb, 'shell', 'am', 'start', '-n', 'jsettlers.main.android/jsettlers.main.android.JsettlersActivity'
+	    }
+	    println 'The application should have started now.'
+	}
 }
-
 
 
 
@@ -137,15 +139,17 @@ task releaseDedicatedServer(type: Copy) {
 }
 
 
-task releaseAndroid(type: Zip, dependsOn: ':jsettlers.main.android:assembleDebug') {
-    into('JSettlers-Android') {
-        from { project.getRootProject().file('LICENSE.txt') }
-        from file(project(':jsettlers.main.android').getBuildDir().absolutePath + '/outputs/apk/jsettlers.main.android-debug.apk')
-        rename 'jsettlers.main.android-debug.apk', 'JSettlers-Android.apk'
-    }
+if (findProject(':jsettlers.main.android')) {
+	task releaseAndroid(type: Zip, dependsOn: ':jsettlers.main.android:assembleDebug') {
+	    into('JSettlers-Android') {
+	        from { project.getRootProject().file('LICENSE.txt') }
+	        from file(project(':jsettlers.main.android').getBuildDir().absolutePath + '/outputs/apk/jsettlers.main.android-debug.apk')
+	        rename 'jsettlers.main.android-debug.apk', 'JSettlers-Android.apk'
+	    }
 
-    destinationDir releaseDir
-    baseName "JSettlers-Android"
+	    destinationDir releaseDir
+	    baseName "JSettlers-Android"
+	}
 }
 
 

--- a/build.gradle
+++ b/build.gradle
@@ -94,9 +94,13 @@ task runMapCreator(dependsOn: ':jsettlers.mapcreator:run') {}
 if (findProject(':jsettlers.main.android')) {
 	// Android is present.
 	task runAndroid(dependsOn: ':jsettlers.main.android:assembleDebug') << {
-	    Properties local = new Properties()
-	    local.load(new FileInputStream("local.properties"))
-	    def adb = local.getProperty('sdk.dir') + '/platform-tools/adb'
+		def androidHome = System.env.ANDROID_HOME;
+		if (!androidHome) {
+		    Properties local = new Properties()
+		    local.load(new FileInputStream("local.properties"))
+			androidHome = local.getProperty('sdk.dir');
+		}
+	    def adb = androidHome + '/platform-tools/adb'
 	    println 'Copy files to your device...'
 	    exec {
 	        commandLine adb, 'uninstall', 'jsettlers.main.android'

--- a/settings.gradle
+++ b/settings.gradle
@@ -20,7 +20,7 @@ include ':jsettlers.tools'
 // Android projects
 // Android is automatically disabled if SDK is not set.
 def androidFile = new File(rootProject.dir, 'local.properties');
-if (androidFile.isFile() && androidFile.text.matches('(?s)^(.*\\n)?\\s*sdk.dir=.*'))  {
+if (System.env.ANDROID_HOME || androidFile.isFile() && androidFile.text.matches('(?s)^(.*\\n)?\\s*sdk.dir=.*'))  {
 	include ':go.graphics.android'
 	include ':jsettlers.graphics.androidui'
 	include ':jsettlers.main.android'

--- a/settings.gradle
+++ b/settings.gradle
@@ -3,22 +3,32 @@
 gradle.ext.projectsWithExtraTest = []
 
 include ':go.graphics'
-include ':go.graphics.android'
 include ':go.graphics.swing'
 include ':jsettlers.buildingcreator'
 include ':jsettlers.common'
 include ':jsettlers.graphics'
 include ':jsettlers.graphics:layoutbuilder'
-include ':jsettlers.graphics.androidui'
 include ':jsettlers.graphics.swing'
 include ':jsettlers.logic'
-include ':jsettlers.main.android'
 include ':jsettlers.main.swing'
 include ':jsettlers.mapcreator'
 include ':jsettlers.network'
 include ':jsettlers.tests'
 include ':jsettlers.testutils'
 include ':jsettlers.tools'
+
+// Android projects
+// Android is automatically disabled if SDK is not set.
+def androidFile = new File(rootProject.dir, 'local.properties');
+if (androidFile.isFile() && androidFile.text.matches('(?s)^(.*\\n)?\\s*sdk.dir=.*'))  {
+	include ':go.graphics.android'
+	include ':jsettlers.graphics.androidui'
+	include ':jsettlers.main.android'
+} else {
+	println 'WARNING: Android is disabled.'
+	println 'To enable Android build, download the Android SDK. Then create a file local.properties with the following content:'
+	println '    sdk.dir=<path to sdk>'
+}
 
 rootProject.name = 'settlers-remake'
 // This will loop through the object descriptors

